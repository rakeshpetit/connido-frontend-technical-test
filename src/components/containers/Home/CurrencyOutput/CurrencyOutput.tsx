import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import moment from 'moment';

type Props = {
  lastUpdatedTimeStamp: number,
  input: string,
  output: string
};

class CurrencyOutput extends Component<Props> {
  render() {
    const { input, output, lastUpdatedTimeStamp } = this.props;
    if (lastUpdatedTimeStamp === undefined) {
      return undefined;
    }
    const lastUpdatedTime = moment.unix(this.props.lastUpdatedTimeStamp).format('DD/MM/YYYY h:mm:ss A');
    return (
      <View style={styles.outer}>
        <View style={styles.inner}>
          <Text style={styles.textInput}>{input}</Text>
          <Text style={styles.textInput}>{output}</Text>
        </View>
        <Text>Last updated API data on {lastUpdatedTime}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  outer: {
    alignItems: 'center',
    backgroundColor: '#F7F9E0'
  },
  inner: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    width: '100%',
    backgroundColor: '#F7F9E0',
    padding: 10
  },
  textInput: {
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 20,
    borderBottomColor: '#000000',
    borderWidth: 1,
    backgroundColor: 'white',
    padding: 10,
    fontSize: 20,
    width: '45%'
  }
});

export default CurrencyOutput;
