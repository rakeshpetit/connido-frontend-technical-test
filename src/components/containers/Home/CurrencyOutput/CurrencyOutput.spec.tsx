import React from 'react';
import renderer from 'react-test-renderer';

import CurrencyOutput from './CurrencyOutput';

it('renders CurrencyOutput correctly', () => {
  const lastUpdatedTimeStamp = 24436453;
  const input = '1';
  const output = '0.8945';
  const tree = renderer.create
    (<CurrencyOutput
        lastUpdatedTimeStamp={lastUpdatedTimeStamp}
        input={input}
        output={output}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});