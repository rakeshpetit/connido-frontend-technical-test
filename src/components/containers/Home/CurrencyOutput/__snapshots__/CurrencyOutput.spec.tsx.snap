// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F9E0",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F7F9E0",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomColor": "#000000",
          "borderWidth": 1,
          "fontSize": 20,
          "marginBottom": 20,
          "marginTop": 20,
          "padding": 10,
          "width": "45%",
        }
      }
    >
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomColor": "#000000",
          "borderWidth": 1,
          "fontSize": 20,
          "marginBottom": 20,
          "marginTop": 20,
          "padding": 10,
          "width": "45%",
        }
      }
    >
      0.8945
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Last updated API data on 
    11/10/1970 1:24:13 AM
  </Text>
</View>
`;
